src/cicd/actions.ts

```diff
@@ -0,0 +1,245 @@
+import { Repository } from 'aws-cdk-lib/aws-codecommit';
+import { PolicyStatement } from 'aws-cdk-lib/aws-iam';
+import {
+  CodeBuildStep,
+  CodePipelineSource,
+  ConnectionSourceOptions,
+  IFileSetProducer,
+  ShellStep,
+} from 'aws-cdk-lib/pipelines';
+import { Construct } from 'constructs';
+import { getCodeartifactPublishPolicyStatements } from './utils';
+
+export interface GetSynthActionProps {
+  readonly codePipelineSource?: IFileSetProducer;
+  readonly cdkVersion?: string;
+  readonly partition?: string;
+  readonly region?: string;
+  readonly account?: string;
+  readonly rolePolicyStatements?: PolicyStatement[];
+  readonly codeartifactRepository?: string;
+  readonly codeartifactDomain?: string;
+  readonly codeartifactDomainOwner?: string;
+}
+
+export interface CodeCommitSourceActionProps {
+  readonly scope: Construct;
+  readonly repositoryName: string;
+  readonly branch: string;
+  readonly props?: ConnectionSourceOptions;
+}
+
+export function getCodeCommitSourceAction(props: CodeCommitSourceActionProps) {
+  return CodePipelineSource.codeCommit(
+    Repository.fromRepositoryName(
+      props.scope,
+      props.repositoryName,
+      props.repositoryName,
+    ),
+    props.branch,
+    props.props,
+  );
+}
+
+export function getSynthAction(props: GetSynthActionProps) {
+  var installCommands;
+  installCommands = [
+    `npm install -g aws-cdk@${props.cdkVersion ? props.cdkVersion : ''}`,
+  ];
+
+  // if (all([codeArtifactRepository, codeArtifactDomain, codeArtifactDomainOwner])) {
+  //   if (!rolePolicyStatements) {
+  //     rolePolicyStatements = _get_codeartifact_read_policy_statements(partition, region, account, codeArtifactDomain, codeArtifactRepository);
+  //   }
+
+  //   install_commands.psuh(`aws codeartifact login --tool pip --repository ${codeArtifactRepository} --domain ${codeArtifactDomain} --domain-owner ${codeArtifactDomainOwner}`);
+  // }
+
+  installCommands.push('pip install -r requirements.txt');
+  return new CodeBuildStep('Synth', {
+    input: props.codePipelineSource,
+    installCommands: installCommands,
+    commands: ['cdk synth'],
+    rolePolicyStatements: props.rolePolicyStatements,
+  });
+}
+
+export interface CfnNagActionProps {
+  readonly fileSetProducer: IFileSetProducer;
+  readonly stageName?: string;
+}
+
+export function getCfnNagAction(props: CfnNagActionProps) {
```

Comment:
I'm trying to figure out whether it's best to use a props interface here, or just have two arguments, like:
```
function getCfnNagAction(fileSetProducer: IFileSetProducer, stageName: string = 'CFNNag') {}
```

When working with CDK, we always pass `props` to CDK Contructs as interfaces. However, I'm not sure if this should also be done for all our internal helper functions. I can't find any documentation or code conventions for this. So I think it's going to be up to us to decide how to handle these cases.

Reply:
I see, I was glibly creating interfaces, but I could see why they might be a bit unnecessary for private methods, i'd vote we stick to just constructs for now.